pkg/ipam/node_manager.go

```diff
@@ -119,6 +119,14 @@ type AllocationImplementation interface {
 	// also called when the IPAM layer detects that state got out of sync.
 	Resync(ctx context.Context) time.Time
 
+	// SupportInstanceSync returns whether the instance incremental sync is
+	// supported. If not, full Resync is called instead.
+	SupportInstanceSync() bool
+
+	// InstanceSync is called to sync the state of the specified instance with
+	// external APIs or systems.
+	InstanceSync(ctx context.Context, instanceID string) time.Time
+
```

Comment:
Looking at this a bit more, it feels a bit cumbersome for the node manager to have to branch based on what the underlying instance API implementation does.

At the end of the day, the IPAM manager just wants to request a resync of the instance state, having a function that returns whether the implementation supports a feature feels like we're leaking implementation details a bit.  

I wonder if this optimization couldn't be done more opaquely by the underlying implementations?